packages/synthetics-sdk-broken-links/test/unit/link_utils.spec.ts

```diff
@@ -214,31 +235,125 @@ describe('GCM Synthetics Broken Links Utilies', async () => {
+      const expectedOutput = "test_@_$%^&_()__+-=__{};'___,.______";
       expect(sanitizeObjectName(input)).to.equal(expectedOutput);
     });
 
     it('should replace the forbidden prefix', () => {
-      const input = ".well-known/acme-challenge/test";
-      const expectedOutput = "_test";
+      const input = '.well-known/acme-challenge/test';
+      const expectedOutput = '_test';
       expect(sanitizeObjectName(input)).to.equal(expectedOutput);
     });
 
     it('should handle standalone "." and ".."', () => {
-      expect(sanitizeObjectName(".")).to.equal("_");
-      expect(sanitizeObjectName("..")).to.equal("_");
+      expect(sanitizeObjectName('.')).to.equal('_');
+      expect(sanitizeObjectName('..')).to.equal('_');
     });
 
     it('should handle null and undefined', () => {
-      expect(sanitizeObjectName(null)).to.equal("_");
-      expect(sanitizeObjectName(undefined)).to.equal("_");
-    })
+      expect(sanitizeObjectName(null)).to.equal('_');
+      expect(sanitizeObjectName(undefined)).to.equal('_');
+    });
 
     it('should trim leading and trailing whitespace', () => {
-      const input = "  test name  ";
-      const expectedOutput = "test_name";
+      const input = '  test name  ';
+      const expectedOutput = 'test_name';
       expect(sanitizeObjectName(input)).to.equal(expectedOutput);
     });
   });
+
+  describe('shouldTakeScreenshot', () => {
+    describe('screenshot_condition: ALL', () => {
+      it('should return true when passed is true', () => {
+        const options = {
+          screenshot_options: { capture_condition: ApiCaptureCondition.ALL },
+        } as BrokenLinksResultV1_BrokenLinkCheckerOptions;
+        const result = shouldTakeScreenshot(options, true);
+        expect(result).to.be.true;
+      });
+
+      it('should return true when passed is false', () => {
+        const options = {
+          screenshot_options: { capture_condition: ApiCaptureCondition.ALL },
+        } as BrokenLinksResultV1_BrokenLinkCheckerOptions;
+        const result = shouldTakeScreenshot(options, false);
+        expect(result).to.be.true;
+      });
+    });
+    describe('screenshot_condition: FAILING', () => {
+      it('should return true if passed is false', () => {
+        const options = {
+          screenshot_options: {
+            capture_condition: ApiCaptureCondition.FAILING,
+          },
+        } as BrokenLinksResultV1_BrokenLinkCheckerOptions;
+        const result = shouldTakeScreenshot(options, false);
+        expect(result).to.be.true;
+      });
+
+      it('should return false if passed is true', () => {
+        const options = {
+          screenshot_options: {
+            capture_condition: ApiCaptureCondition.FAILING,
+          },
+        } as BrokenLinksResultV1_BrokenLinkCheckerOptions;
+        const result = shouldTakeScreenshot(options, true);
+        expect(result).to.be.false;
+      });
+    });
+
+    describe('screenshot_condition: NONE', () => {
```

Comment:
Nit on consistency here, if we can combine the NONE case into 1 test then we should do it for the others, or split up the NONE case into 2 tests